synapse/storage/database.py

```diff
@@ -2059,6 +2065,8 @@ def simple_update_many_txn(
             raise ValueError(
                 f"{len(key_values)} key rows and {len(value_values)} value rows: should be the same number."
             )
+        if not key_values:
```

Comment:
I think we should also check the `value_values`? I *think* it's probably fine, but we do specifically handle the case of an empty `key_names` below.

Reply:
Ah, didn't realise they needed to be the same length